docs/yaml/elementary/dict.yml

```diff
@@ -8,8 +8,9 @@ description: |
   You can also iterate over dictionaries with the [`foreach`
   statement](Syntax.md#foreach-statements).
 
-  *(since 0.48.0)* Dictionaries can be added (e.g. `d1 = d2 + d3` and `d1 += d2`).
+  *(since 0.48.0)*: Dictionaries can be added (e.g. `d1 = d2 + d3` and `d1 += d2`).
   Values from the second dictionary overrides values from the first.
+  *(since 0.62.0)*: Dictionary order is guaranteed to be insertion order.
```

Comment:
From the commit message:
> This is a side effect of requiring Python >= 3.7 which itself guarantees dictionary order. This is now a Meson language guarantee as well which is required for passing default_options as dict and is generally expected by users.

Just curious why this is a requirement for default_options?

Reply:
In the case you mix buildtype and debug/optimization, the order between them makes a difference.

Reply:
Hmm, I kind of feel like the answer should be "don't do that". :P

I would typically only do that in contexts where the arguments are layered or composed, e.g. on the command line from a script. But to each their own...

Reply:
In theory can also matter in cases like:
```meson
d = {'foo': true}
d += {'foo': false}
library(..., default_options: d)
```

But that's niche corner case too.

Reply:
Yes, it's niche and you probably shouldn't do that anyway. I think it's generally useful to guarantee dict order, especially since we have an API that really suggest there is an order `d += {foo: bar}`. I wouldn't be surprised there are projects already relying on that implicitly.

Reply:
@eli-schwartz because our stringlist implementation has the behavior of being ordered. Without that guarantee you could get different behavior between writing `deafult_options : ['a=b', 'a=c']` and `default_options : {'a' :'b', 'a' : 'c'}`. Even if you maybe shouldn't do that.

Reply:
@dcbaker you would get an error about duplicated keys in dict in that case.